src/javascript/app_2/App/Components/Elements/AccountSwitcher/account_switcher.jsx

```diff
@@ -105,10 +94,18 @@ class AccountSwitcher extends React.Component {
 }
 
 AccountSwitcher.propTypes = {
+    client            : PropTypes.object,
     is_upgrade_enabled: PropTypes.bool,
     is_visible        : PropTypes.bool,
+    modules           : PropTypes.object,
     onClickUpgrade    : PropTypes.func,
     toggle            : PropTypes.func,
 };
 
-export { AccountSwitcher };
+const account_switcher = connect(
```

Comment:
@easteregg 
Regarding the [StyleGuid](https://github.com/binary-com/binary-static/blob/new-app/src/javascript/app_2/Documents/STYLE_GUIDE.MD) just container are allowed to connect to stores.

Reply:
I think this component has more than enough responsibilities to be considered a presentational component. already it creates a list for the user (via direct calls to utilities). the nearest container component (`header.jsx`) is far up on the tree and it just creates a chain of props that are just being passed down the chain to get to this component. It does not result in a cleaner code IMO, nor it separates the concerns.